src/Corax/IndexWriter.cs

```diff
@@ -387,29 +386,32 @@ private unsafe void DeleteCommit(Span<byte> tmpBuf, Tree fieldsTree)
                             while (it.ReadNext())
                             {
                                 if (it.IsNull)
-                                    DeleteToken(Constants.NullValueSlice.AsSpan(), tmpBuf, binding, tempWordsSpace, tempTokenSpace);
+                                    DeleteIdFromTerm(Constants.NullValueSlice.AsSpan(), tmpBuf, binding, tempWordsSpace, tempTokenSpace);
                                 else
-                                    DeleteToken(it.Sequence, tmpBuf, binding, tempWordsSpace, tempTokenSpace);
+                                    DeleteIdFromTerm(it.Sequence, tmpBuf, binding, tempWordsSpace, tempTokenSpace);
                             }
                         }
                     }
                     else
                     {
                         entryReader.Read(binding.FieldId, out var termValue);
-                        DeleteToken(termValue, tmpBuf, binding, tempWordsSpace, tempTokenSpace);
+                        DeleteIdFromTerm(termValue, tmpBuf, binding, tempWordsSpace, tempTokenSpace);
                     }
                 }
+                
+                Container.Delete(llt, _entriesContainerId, entryToDelete); // delete raw index entry
+                llt.RootObjects.Increment(Constants.IndexWriter.NumberOfEntriesSlice, -1); // update number of entries
```

Comment:
Why call it each time and not once after the loop?

Reply:
@redknightlois ping